src/create.jl

```diff
@@ -1,41 +1,344 @@
+const TypeNamedTuple = NamedTuple{<:Any,<:Tuple{Vararg{Type}}}
 
+"""
+    create([f!], [filename], template; kw...)
```

Comment:
When should I use this as opposed to `Raster(data, dims; ...)`?  What are the advantages beyond efficiency in writing to file?

Reply:
Or is the direct constructor going to call `create`?

Reply:
Well `Raster(filename)` reads an existing file... create creates one. 

So this is best for building something new from scratch, `Raster` is best for opening something, rewrapping a DimArray, etc

Reply:
Gotcha, I guess I'm not entirely clear on what to use when creating a purely in memory dataset.  Does the pure `Raster(data, dims; kwargs...)` constructor have the same options?  Should it?   That's the most intuitive option IMO to create an in memory raster...